leetcode/BinaryTreeTests/BinarySearchTree_1038.cs

```diff
@@ -0,0 +1,52 @@
+﻿namespace BinaryTreeTests;
+
+[TestFixture]
+class BinarySearchTree_1038
+{
+    private class Solution
+    {
+        public TreeNode BstToGst(TreeNode root)
+        {
+            var sum = 0;
+            return InorderRight(root);
+
+            TreeNode InorderRight(TreeNode node)
+            {
+                if (node is null) return null;
+                var right = InorderRight(node.right);
+                sum += node.val;
+                var sumNode = new TreeNode(sum);
+                var left = InorderRight(node.left);
```

Comment:
With correct traverse order and tracked `sum` we can easily build Greater Sum Tree within 1 iteration